### PR TITLE
A couple charge job refactors

### PR DESCRIPTION
**Pull request by craigspaeth**
This does a couple small things related to data consistency I encountered in the charge job—I'll comment more inline.

### CODE REVIEW

File: apps/api/models/patient.js [hunk 1]
```diff
@@ -10,7 +10,9 @@ const { REFERRAL_NOTE_TEMPLATE_ID } = process.env
 export const fromCliniko = patient => ({
   name: [patient.first_name, patient.last_name].join(' '),
   email: patient.email,
-  phone: patient.patient_phone_numbers[0].number
+  phone: patient.patient_phone_numbers.length
```
Comment (craigspaeth) [inline apps/api/models/patient.js]: Apparently, some patients don't have phone numbers in Cliniko. I think we assumed this was a requirement in our product but truth is we probably have folks being manually input into Cliniko and we probably shouldn't expect to have anything but an email—so this makes it a bit more defensive.

File: apps/jobs/lib/charge-sessions/network.js [hunk 2]
```diff
@@ -34,56 +37,70 @@ export const addBillableDataToAppts = async appts => {
...
-        historicalCharges,
-        invoiceContent,
-        invoiceNotification,
-        lead,
-        patient,
-        practitioner,
-        stripeCustomer
-      }
-    })
+        if (!billableItem) return null
```
Comment (craigspaeth) [inline apps/jobs/lib/charge-sessions/network.js]: I was often hitting Cliniko rate limiting so I slightly re-ordered the fetching so we can exit early if something required for charging is necessary (e.g. we don't fetch all the other things if the appt doesn't have a billable item).

File: apps/jobs/lib/charge-sessions/network.js [hunk 3]
```diff
@@ -34,56 +37,70 @@ export const addBillableDataToAppts = async appts => {
...
+            customer: stripeCustomer.id
+          })).data
+          : null
+        const lead =
+          (await at.find({
+            table: 'leads',
+            filter: `emailAddress='${patient.email}'`
+          }))[0] ||
+          (await (async () => {
+            await syncPatients([patient])
```
Comment (craigspaeth) [inline apps/jobs/lib/charge-sessions/network.js]: I still need to run the sync patients/lead job across all of our data, so until Airtable/Cliniko are in a consistent state, this code runs lazily here to ensure people who need to be charged get in a consistent state before doing so.

**Comment by lionsdwarf** [comment]
Copy that! Thanks for the updates, Craig-o.

**Pull request merged**